stream_alert/rule_processor/pre_parsers.py

```diff
@@ -79,6 +78,32 @@ def pre_parse_sns(cls, raw_record):
         return base64.b64decode(raw_record['Sns']['Message'])
 
     @classmethod
+    def read_s3_file(cls, downloaded_s3_object):
+        """Parse a downloaded file from S3
+
+        Supports reading both gzipped files and plaintext files.
+
+        Args:
+            downloaded_s3_object (string): A full path to the downloaded file.
+
+        Returns:
+            (list) Lines from the downloaded s3 object
+        """
+        _, extension = os.path.splitext(downloaded_s3_object)
+
+        if extension == '.gz':
+            for line in gzip.open(downloaded_s3_object, 'r'):
+                yield line.rstrip()
+        else:
+            for line in open(downloaded_s3_object, 'r'):
+                yield line.rstrip()
+
+        # remove the file
+        os.remove(downloaded_s3_object)
```

Comment:
This doesn't work as expected in Lambda - you have to truncate the file first to avoid using all of the disk space